### ISSUE TITLE
Should dracut uncompress compressed kernel modules before installing into the initramfs?

### ISSUE THREAD

**Issue by AndrewAmmerlaan** [dracut-ng/dracut-ng#560]
A kernel that is configured to compress modules (for example `CONFIG_MODULE_COMPRESS_XZ=y`) still supports loading uncompressed modules. Given that it is size-wise more efficient to compress a bunch of uncompressed files, compared to compressing a bunch of files that are individually already compressed. It might make sense for Dracut to decompress modules it finds before including them in the initramfs.

The same holds for firmware files (`CONFIG_FW_LOADER_COMPRESS_XZ=y`).

To put this in numbers, an UKI for the same kernel with the same user space utilities, but with pre-xz-compressed kernel modules and firmware is 356.352 kb larger on my system.

**Comment by AndrewAmmerlaan** [comment]
Proof of concept: https://github.com/dracut-ng/dracut-ng/pull/568

**Comment by bdrung** [comment]
The better approach will be https://github.com/dracut-ng/dracut-ng/issues/110 which will reduce the boot time.

**Comment by AndrewAmmerlaan** [comment]
> The better approach will be #110 which will reduce the boot time.

Hmm, yes this might make more sense. Though this would not reduce the overal size of the combined initramfs I think, right?

**Comment by bdrung** [comment]
> > The better approach will be #110 which will reduce the boot time.
> 
> Hmm, yes this might make more sense. Though this would not reduce the overal size of the combined initramfs I think, right?

Yes, the size will roughly stay the same iff the initrd and kernel module compression levels are the same. For example Ubuntu uses the highest compression level for the kernel modules (compressed during package build) and low compression levels for the initrd (because some users machines do not have enough memory for high compression levels).
